include/stf_decoder.hpp

```
@@ -333,6 +333,18 @@ namespace stf {
                 return stf::INST_MEM_ACCESS::INVALID;
             }
 
+            /**
+             * Returns the data size associated with the opcode
+             */
+            inline uint32_t getDataSize() const {
+                try {
+                    return getDecodeInfo_()->opinfo->getDataSize();
+                }
+                catch(const InvalidInstException&) {
```

I suppose other things will warn about these exceptions, so it doesn't seem like something this method needs to handle.  You may want to mention this behavior (ignore exceptions, return 0) in the doxygen comment.